### PR TITLE
feature58/ screening test functionality

### PR DESCRIPTION
**Pull request by AgentQuantum**
# Complete Screening Test Functionality

## WHAT

This PR completes the screening test functionality with full database persistence and participant routing. It ensures all 5 steps of the screening questionnaire work end-to-end, properly save data to the database, and route participants to appropriate tests based on their responses.

**Key Changes:**
- Added database initialization script for proper table creation
- Enhanced backend error handling with rollback for all screening endpoints
- Fixed frontend screening flow to ensure complete data persistence
- Improved participant dashboard with recommended test links
- Added comprehensive error handling for validation and edge cases
- Updated npm scripts for easier development workflow

## WHY

The screening test is a critical component that determines participant eligibility and routes them to appropriate tests. Previous implementation had gaps in error handling and data persistence, which could lead to:
- Incomplete screening data in the database
- Participants not being routed correctly
- Poor error messages causing confusion
- Database initialization issues preventing proper table creation

This PR ensures:
- ✅ All screening data is reliably saved at each step
- ✅ Participants are correctly routed based on their responses
- ✅ Proper error handling prevents data loss
- ✅ Database tables are correctly initialized
- ✅ Better developer experience with improved npm scripts

### Backend
- [x] Database initialization script creates all required tables
- [x] All screening endpoints have proper error handling with rollback
- [x] Participant age validation handles empty strings correctly
- [x] Dashboard correctly queries ScreeningRecommendedTest table (normalized)
- [x] Invalid roles are handled gracefully in authentication endpoints

### Frontend
- [x] All 5 screening steps navigate correctly
- [x] Step 4 always saves data before routing (even if no eligible types)
- [x] ScreeningExit page displays appropriate messages for each exit code
- [x] ParticipantDashboard shows recommended tests as clickable links
- [x] Error messages are displayed to users when API calls fail

### End-to-End Flow
- [x] Complete screening flow works without errors
- [x] Data persists correctly in database at each step
- [x] Participants are routed to exit pages or step 5 summary correctly
- [x] Recommended tests are generated based on participant responses

### Testing
- [x] Unit tests added for useScreeningState hook
- [x] Integration tests added for ScreeningFlow component
- [x] Test setup configuration in place

## How to Run

**Backend Setup:**
```bash
cd api
python -m venv .venv
.venv\Scripts\activate  # Windows (or source .venv/bin/activate on Mac/Linux)
pip install -r requirements.txt
python init_db.py  # IMPORTANT: Initialize database first
cd ..
```

**Frontend Setup:**
```bash
npm install
```

**Run Application:**
```bash
npm start  # Runs both backend (port 5000) and frontend (port 5173) concurrently
```

**Or run separately:**
```bash
npm run api       # Terminal 1: Backend
npm run dev      

Closes issue #59

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,22 +60,38 @@ alembic/
...
 
 # ======================================
-# Node / Frontend (if used)
+# Node / Frontend
 # ======================================
 node_modules/
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+package-lock.json
```
Comment (ngvt1n) [inline .gitignore]: hello, package-lock should not be gitignored. we have to version-control the dependency tree just as much as our code

File: package.json [hunk 2]
```diff
@@ -5,27 +5,37 @@
...
+    "dev:frontend": "vite",
+    "dev:backend": "cd api && python app.py",
+    "dev:all": "concurrently -n \"backend,frontend\" -c \"yellow,blue\" \"npm run dev:backend\" \"npm run dev:frontend\"",
+    "start": "npm run dev:all",
     "build": "vite build",
     "preview": "vite preview",
     "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0",
-    "api": "cd api && .venv\\Scripts\\flask run --no-debugger"
+    "test": "vitest run",
+    "api": "cd api && python app.py",
```
[the remark this answers is not in the document]
Comment (ngvt1n) [reply package.json]: i guess this duplicates (npm run api and npm run dev:backend does the same thing, but we can keep for backwards compatibility)

File: .gitignore (hunk 1 above)
Comment (AgentQuantum) [reply .gitignore]: Fixed
